executor/read_only_checker.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/pingcap/tidb/ast"
+)
+
+// IsReadOnly checks whether the input ast is readOnly.
+func IsReadOnly(node ast.Node) bool {
```

Comment:
maybe a better way is put it into package `ast` or `plan` ?

Reply:
Yes, I think package ```ast``` is a better place